File: backend/internal/domain/entities/address.go
package entities

type Address struct {
	CEP          string  `json:"cep"`
	Street       string  `json:"street"`
	Number       *string `json:"number,omitempty"`
	Complement   *string `json:"complement,omitempty"`
	Neighborhood string  `json:"neighborhood"`
	City         string  `json:"city"`
	State        string  `json:"state"`
}

File: frontend/lib/modules/listing/domain/entities/address.dart
class Address {
  const Address({
    this.cep = '',
    this.street = '',
    this.number,
    this.complement,
    this.neighborhood = '',
    this.city = '',
    this.state = '',
  });

  final String cep;
  final String street;
  final String? number;
  final String? complement;
  final String neighborhood;
  final String city;
  final String state;

  bool get isComplete =>
      street.isNotEmpty &&
      neighborhood.isNotEmpty &&
      city.isNotEmpty &&
      state.isNotEmpty;
}
